app/controllers/teachers_controller.rb

```diff
@@ -0,0 +1,54 @@
+class TeachersController < ApplicationController
+  def show
+  end
+
+  def edit
+    if params[:update] && params[:update] == "confirm"
+      @edit_fields = ["Old Password","New Password", "Confirm New Password"]
+    elsif params[:update] && params[:update] == "reset"
+      @edit_fields = ["New Password", "Confirm New Password"]
+    else
+      @edit_fields = ["first_name", "last_name", "email"]
+    end
+  end
+
+  def update
```

Comment:
This method is pretty long.

I wonder if it might be worth creating an issue for breaking this functionality into the `update` actions for some single-responsibility controllers?

It also looks like the redirect is hit whether or not there is success or failure.
I see from the note on this PR that there is still work being done on this logic